src/ui/public/kuery/suggestions/value.js

```diff
@@ -1,16 +1,26 @@
+import 'isomorphic-fetch';
 import { flatten, memoize } from 'lodash';
 import chrome from '../../chrome';
 import { escapeQuotes } from './escape_kuery';
 
 const baseUrl = chrome.addBasePath('/api/kibana/suggestions/values');
 const type = 'value';
 
-export function getSuggestionsProvider({ $http, config, indexPatterns }) {
+const requestSuggestions = memoize(async (query, field) => {
+  const response = await fetch(`${baseUrl}/${field.indexPattern.title}`, {
+    method: 'POST',
+    body: JSON.stringify({ query, field: field.name }),
+    headers: new Headers({
+      'Content-Type': 'application/json',
+      'kbn-xsrf': true
+    }),
+  });
+  return response.json();
+}, resolver);
+
+
+export function getSuggestionsProvider({ config, indexPatterns }) {
   const allFields = flatten(indexPatterns.map(indexPattern => indexPattern.fields.raw));
```

Comment:
Are you going to want to change this here to use `getFieldsByName` as well? It might cause problems where we use `field.indexPattern.title`, but we could switch this around so that the `requestSuggestions` function can accept an `indexPattern` object as well, or just a `title` since that's all it's using anyway.

Reply:
Btw. `indexPatterns` is an array of `indexPattern` objects that have the structure `{field: [...], title: 'foo-*'}`